runtime-modules/content/src/migrations.rs

```diff
@@ -0,0 +1,219 @@
+            // ChangeChannelFeatureStatus can contain all possible PausableChannelFeature variants
+            .saturating_add((PausableChannelFeature::VARIANT_COUNT as u32).saturating_sub(1))
+            // DeleteVideoAssets can contain `true` or `false`
+            .saturating_add(1);
+    }
+
+    #[derive(Encode, Decode, Eq, PartialEq, Clone, Debug, TypeInfo, MaxEncodedLen)]
+    struct CuratorGroupRecordV0<CuratorGroupCuratorsMap, ModerationPermissionsByLevel> {
+        /// Map from CuratorId to curator's ChannelAgentPermissions
+        pub curators: CuratorGroupCuratorsMap,
+
+        /// When `false`, curator in a given group is forbidden to act
+        pub active: bool,
+
+        // Group's moderation permissions (by channel's privilage level)
+        pub permissions_by_level: ModerationPermissionsByLevel,
+    }
+
+    type CuratorGroupV0<T> =
+        CuratorGroupRecordV0<CuratorGroupCuratorsMap<T>, StoredModerationPermissionsByLevelV0<T>>;
+
+    type StoredCuratorModerationPermissionsV0 =
+        BoundedBTreeSet<iterable_enums::ContentModerationActionV0, MaxCuratorPermissionsPerLevelV0>;
+
+    type StoredModerationPermissionsByLevelV0<T> = BoundedBTreeMap<
+        <T as Config>::ChannelPrivilegeLevel,
+        StoredCuratorModerationPermissionsV0,
+        <T as Config>::MaxKeysPerCuratorGroupPermissionsByLevelMap,
+    >;
+
+    // #[frame_support::storage_alias]
+    // type CuratorGroupByIdV0<T: Config> = StorageMap<
+    //     Pallet<T>,
+    //     Blake2_128Concat,
+    //     <T as permissions::ContentActorAuthenticator>::CuratorGroupId,
+    //     CuratorGroupV0<T>,
+    // >;
+
+    pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
+    impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+            let onchain = Pallet::<T>::on_chain_storage_version();
+
+            ensure!(onchain < 1, "this migration can be deleted");
+
+            // using storage alias CuratorGroupByIdV0 was not working.
+            // most likely wrong prefix is getting used, since even iterating over keys return 0 count.
+            // So we use the new map but only iterate over keys to avoid decode errors. (if decode fails the item is skipped)
+            let group_count: usize = CuratorGroupById::<T>::iter_keys().count();
+            log!(info, "Number of curator groups pre_upgrade {}", group_count);
+
+            Ok(group_count.to_be_bytes().to_vec())
+        }
+
+        fn on_runtime_upgrade() -> Weight {
+            let onchain = Pallet::<T>::on_chain_storage_version();
+            let current = Pallet::<T>::current_storage_version();
+
+            if onchain > 0 {
+                return T::DbWeight::get().reads(1);
+            }
+
+            use iterable_enums::ContentModerationActionV0;
+
+            <CuratorGroupById<T>>::translate_values(
```

Comment:
Are we not concerned about this, for some reason being gigantic by some weird accident or attack, and then trying to iterate it all in one block? Seems like this iteration must be either unrolled across blocks if it is truly to be dynamic, or it has to have a hardcap which we know allows compute to finish in time.

Reply:
That is a great point, I guess it also goes back to the discussion here https://github.com/Joystream/joystream/pull/4874#discussion_r1344448892

I believe there is already a limit on the number of curators per group but not on number of groups.
So yes it could be a problem if the content lead decides to create a very large number of groups before the runtime upgrade.

Should we tackle this possibility ?